### ISSUE TITLE
Strange Problem with header colum names

### ISSUE THREAD

**Issue by oderwat** [harelba/q#79]
Can anybody explain if this is a bug or I am doing something wrong?

> head data.csv

```
laid,cnt,zuletzt,frei
0,0,2001-01-01,J
1,1074,2014-12-10,-
2,1092,2014-12-10,-
3,19,2013-12-03,-
4,87,2012-08-06,-
5,193,2014-03-08,-
6,7,2014-12-08,-
7,444,2014-12-10,-
8,266,2014-07-22,-
```

> q -A -H -d , data.csv

```
Table for file: data.csv
  `laid` - int
  `cnt` - int
  `zuletzt` - text
  `frei` - text
```

But:

> q -H -d , "SELECT \* FROM data.csv WHERE laid = 3"

```
query error: no such column: laid
Warning - There seems to be a "no such column" error, and -H (header line) exists. Please make sure that you are using the column names from the header line and not the default (cXX) column names
```

I just don't understand it...

I am using homebrew version 1.4.0 of the tool.

**Comment by cstork** [comment]
I just experienced the same problem as the OP and it took me some time to realize what the problem was and to find this issue including the proposed solution. 

My file was generated by Excel when I saved to a .csv file, which leads me to believe that this must be a relatively common issue. Since a real solution to this problem seems to be more involved (e.g. #217) I think future users would really appreciate a little mention of this failure mode in the error message!  Maybe something like

```
query error: no such column ...
Warning - There seems to be a "no such column" error, and -H (header line) exists. 
Please make sure that you are using the column names from the header line and 
not the default (cXX) column names. This might also be caused by a byte order mark (BOM).
```
?
